src/main/kotlin/com/hivemq/extension/gradle/RunHivemq.kt

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020-present HiveMQ and the HiveMQ Community
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hivemq.extension.gradle
+
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.JavaExec
+
+/**
+ * Runs HiveMQ.
+ *
+ * @author Silvio Giebl
+ */
+@Suppress("LeakingThis")
```

Comment:
Is this suppression necessary?

Reply:
Yes, because the task needs to be non-final (because Gradle generates a dynamic subclass) and methods are called in the constructor which might lead to leaking this if the methods are overwritten.
But this is not the case with the Gradle generated subclass, so this is safe.